x/programs/examples/nft.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package examples
+
+import (
+	"context"
+	"github.com/ava-labs/avalanchego/utils/logging"
+
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/x/programs/runtime"
+	"github.com/ava-labs/hypersdk/x/programs/utils"
+
+	"go.uber.org/zap"
+)
+
+func NewNFT(log logging.Logger, programBytes []byte, maxFee uint64, costMap map[string]uint64, m Metadata) *NFT {
+	return &NFT{
+		log:          log,
+		programBytes: programBytes,
+		metadata:     m,
+		maxFee:       maxFee,
+		costMap:      costMap,
+	}
+}
+
+type NFT struct {
+	log          logging.Logger
+	programBytes []byte
+	metadata     Metadata
+	// metering
+	maxFee  uint64
+	costMap map[string]uint64
+}
+
+type Metadata struct {
+	Name   string
+	Symbol string
+	URI    string
+}
+
+func (t *NFT) Run(ctx context.Context) error {
```

Comment:
We should migrate this into simulator YAML + rust-based tests (wraps contract).

This is blocked by: #537